packages/react-instantsearch-core/src/server/getServerState.tsx

```diff
@@ -127,7 +134,7 @@ function execute({
         );
       }
 
-      return waitForResults(searchRef.current);
+      return waitForResults(searchRef.current, skipRecommend);
```

Comment:
if we're skipping recommend here, where are we actually getting the recommend results?

Reply:
It's in the `initialResults` passed from the first render